presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.sql.analyzer.Analysis;
+import com.facebook.presto.sql.analyzer.TypeSignatureProvider;
+import com.facebook.presto.sql.tree.ArrayConstructor;
+import com.facebook.presto.sql.tree.Cast;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.GroupingOperation;
+import com.facebook.presto.sql.tree.LongLiteral;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.QuerySpecification;
+import com.facebook.presto.sql.tree.SimpleGroupBy;
+import com.facebook.presto.type.ListLiteralType;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.operator.scalar.GroupingOperationFunction.GROUPING;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.analyzer.ExpressionAnalyzer.resolveFunction;
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableList;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public final class GroupingOperationRewriter
+{
+    private GroupingOperationRewriter() {}
+
+    public static Expression rewriteGroupingOperation(GroupingOperation expression, QuerySpecification queryNode, Analysis analysis, Metadata metadata, Optional<Symbol> groupIdSymbol)
+    {
+        requireNonNull(queryNode, "node is null");
+        requireNonNull(analysis, "analysis is null");
+        requireNonNull(metadata, "metadata is null");
+        requireNonNull(groupIdSymbol, "groupIdSymbol is null");
+
+        checkState(queryNode.getGroupBy().isPresent(), "GroupBy node must be present");
+
+        // No GroupIdNode and a GROUPING() operation imply a single grouping, which
+        // means that any columns specified as arguments to GROUPING() will be included
+        // in the group and none of them will be aggregated over. Hence, re-write the
+        // GroupingOperation to a constant literal of 0.
+        // See SQL:2011:4.16.2 and SQL:2011:6.9.10.
+        boolean isSimpleGroupByPresent = queryNode.getGroupBy().get().getGroupingElements().stream().anyMatch(SimpleGroupBy.class::isInstance);
```

Comment:
>  For a single GROUPING SET with more than one column we create a GroupIdNode even though it is semantically the same as a single GROUP BY. I'll look for an existing issue addressing this and if I don't find one I'll file it.

I think it may be because how SQL grammar is defined. In any case, the end result should the same, right?